### PR TITLE
Add Guide: Deploying a GitLab Cluster on GCP/GKE

### PR DESCRIPTION
**Pull request by robmorgan**
This PR adds a new guide to our docs site for deploying a production-grade GitLab cluster on GCP/GKE.

### CODE REVIEW

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+}
+
+variable "cluster_service_account_description" {
+  description = "A description of the custom service account used for the GKE cluster."
+  default     = "Example GKE Cluster Service Account managed by Terraform"
+}
+
+variable "node_machine_type" {
+  description = "The machine type to use for GKE cluster nodes, see https://cloud.google.com/kubernetes-engine/pricing for more details"
+  default     = "db-n1-standard-1"
```
Comment (autero1) [inline content/guides/deploying-a-gitlab-cluster-on-gke/index.md]:
```suggestion
  default     = "n1-standard-1"
```

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+  # 'master_user_name'@'master_user_host' IDENTIFIED BY 'master_user_password'.
+  # These should typically be set as the environment variable TF_VAR_db_master_user_password, etc.
+  # so you don't check these into source control."
+  master_user_password = "${var.db_master_user_password}"
+
+  master_user_name = "${var.db_master_user_name}"
+  master_user_host = "%"
+
+  # Pass the private network link to the module
+  private_network = "${module.vpc_network.private_subnetwork}"
```
Comment (autero1) [inline content/guides/deploying-a-gitlab-cluster-on-gke/index.md]:
```suggestion
  private_network = "${module.vpc_network.network}"
```

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+}
+
+# Reserve global internal address range for the peering
+resource "google_compute_global_address" "private_ip_address" {
+  provider      = "google-beta"
+  name          = "${local.private_ip_name}"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = "${module.vpc_network.private_subnetwork}"
```
Comment (autero1) [inline content/guides/deploying-a-gitlab-cluster-on-gke/index.md]:
```suggestion
  network       = "${module.vpc_network.network}"
```

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md [hunk 4]
```diff
@@ -0,0 +1,541 @@
...
+}
+
+# Establish VPC network peering connection using the reserved address range
+resource "google_service_networking_connection" "private_vpc_connection" {
+  provider                = "google-beta"
+  network                 = "${module.vpc_network.private}"
```
Comment (autero1) [inline content/guides/deploying-a-gitlab-cluster-on-gke/index.md]:
```suggestion
  network                 = "${module.vpc_network.network}"
```

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md [hunk 5]
```diff
@@ -0,0 +1,541 @@
...
+}
+
+variable "postgres_version" {
+  description = "The engine version of the database, e.g. `POSTGRES_9_6`. See https://cloud.google.com/sql/docs/db-versions for supported versions."
+  default     = "POSTGRES_9_6"
+}
+
+variable "db_machine_type" {
+  description = "The machine type to use, see https://cloud.google.com/sql/pricing for more details"
+  default     = "db-n1-standard-1"
```
Comment (autero1) [inline content/guides/deploying-a-gitlab-cluster-on-gke/index.md]:
```suggestion
  # See https://cloud.google.com/sql/docs/postgres/create-instance#machine-types
  default     = "db-custom-1-3840"
```

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md [hunk 6]
```diff
@@ -0,0 +1,541 @@
...
+  default     = "Example GKE Cluster Service Account managed by Terraform"
+}
+
+variable "node_machine_type" {
+  description = "The machine type to use for GKE cluster nodes, see https://cloud.google.com/kubernetes-engine/pricing for more details"
+  default     = "db-n1-standard-1"
+}
+
+# Note, after a db instance name is used, it cannot be reused for up to one week.
+variable "db_name_prefix" {
```
Comment (autero1) [inline content/guides/deploying-a-gitlab-cluster-on-gke/index.md]: This one doesn't have a default - should be moved to REQUIRED or a default added

**Comment by autero1** [comment]
The changes I've suggested should get you through the `terraform` apply with a HA PG DB.

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md [hunk 7]
```diff
@@ -0,0 +1,541 @@
...
+gitlab/gitlab           1.8.4           11.10.4         Web-based Git-repository manager with wiki and issue-trac...
+gitlab/gitlab           1.8.3           11.10.3         Web-based Git-repository manager with wiki and issue-trac...
+```
+
+**Note**: The GitLab chart doesn't have the same version number as GitLab itself. For more information please refer to the [GitLab version mappings](https://docs.gitlab.com/charts/installation/version_mappings.html) document.
+
+Please read the GitLab [Helm Deployment Guide](https://docs.gitlab.com/charts/installation/deployment.html) for more information on customizing
+the GitLab installation.
+
+## Deploying the Dockerized App
```
Comment (yorinasub17) [inline content/guides/deploying-a-gitlab-cluster-on-gke/index.md]: Do we need this section for this doc, when it is already covered in the other doc?

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md [hunk 8]
```diff
@@ -0,0 +1,541 @@
...
+
+- GitLab Pages
+- GitLab Geo
+- No in-cluster HA database
+- Smartcard authentication
+
+We can check the latest versions using the following command:
+
+```bash
+$ helm search -l gitlab/gitlab | head -n4
```
Comment (yorinasub17) [inline content/guides/deploying-a-gitlab-cluster-on-gke/index.md]: This doesn't actually install the chart. Is that intentional?

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md (hunk 6 above)
Comment (robmorgan) [reply content/guides/deploying-a-gitlab-cluster-on-gke/index.md]: Yep agreed

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md (hunk 7 above)
Comment (robmorgan) [reply content/guides/deploying-a-gitlab-cluster-on-gke/index.md]: @yorinasub17 nope its pure copy pasta and I'll remove it plus add the Helm install GitLab chart bit.

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md (hunk 8 above)
Comment (robmorgan) [reply content/guides/deploying-a-gitlab-cluster-on-gke/index.md]: @yorinasub17 yep purely for listing available versions, but I need to add an install section.

File: content/guides/deploying-a-gitlab-cluster-on-gke/index.md (hunk 6 above)
Comment (robmorgan) [reply content/guides/deploying-a-gitlab-cluster-on-gke/index.md]: Ok done.

**Pull request closed without merging**